scripts/check_copyright_notice.py

```diff
@@ -99,10 +112,78 @@
 #   limitations under the License\.
 #
 # ------------------------------------------------------------------------------
-"""
+""",
+    re.MULTILINE,
+)
+
+REGEX_LIST: List[Tuple[str, re.Pattern]] = [
+    ("FetchAI", HEADER_REGEX_FETCHAI),
+    ("Valory", HEADER_REGEX_VALORY),
+    ("Mixed", HEADER_REGEX_MIXED),
+]
+
+
+def _check_copyright(file: Path, match: re.Match) -> Tuple[bool, str]:
+    """
+    Given a file, check if the header stuff is in place.
+
+    Return True if the files has the encoding header and the copyright notice,
+    optionally prefixed by the shebang. Return False otherwise.
 
+    :param file: the file to check.
+    :param match: match object.
+    :return: True if the file is compliant with the checks, False otherwise.
+    """
 
-def check_copyright(file: Path) -> bool:
+    copyright_years_str = match.groups(0)[1]  # type: ignore
+    copyright_years = tuple(int(i) for i in copyright_years_str.split("-"))
+    date_string, _ = subprocess.Popen(  # pylint: disable=consider-using-with  # nosec
+        [str(GIT_PATH), "log", "-1", '--format="%ad"', "--", str(file)],
+        stdout=subprocess.PIPE,
+    ).communicate()
+    date_string_ = date_string.decode().strip()
+    if date_string_ == "":
+        modification_date = datetime.now()
+    else:
+        modification_date = datetime.strptime(date_string_, '"%a %b %d %X %Y %z"')
+
+    # Start year is not 2021 or 2022
+    if copyright_years[0] not in [2018, 2019, 2020, 2021, 2022]:
+        return False, "Start year is not 2021 or 2022."
```

Comment:
for fetch.ai files the start year can be 2019 and 2020 too. 2018 is not possible.